src/bufread.rs

```diff
@@ -110,12 +110,15 @@ impl<R: Read> Read for Cryptostream<R> {
         };
         if eof {
             self.finalized = true;
-            let write_bytes = self
-                .crypter
-                .finalize(&mut buf[bytes_written..])
-                .map_err(|e| Error::new(ErrorKind::Other, e))?;
-            // eprintln!("Wrote {} bytes to encrypted stream", write_bytes);
-            bytes_written += write_bytes;
+
+            if bytes_read > 0 {
```

Comment:
I think this test is too specific and we need to check cumulative `bytes_read` across all previous calls to `read` against the same instance.

A previous call to `read()` could have encrypted a complete block (and consumed/transformed the entirety of the input) but there's no guarantee that `Crypter::finalize()` even after one or more `Crypter::update()` calls at block boundaries is a no-op, as there could be trailers required in the final output (or just system resources that need to be freed by calling `finalize()`).

Reply:
@mqudsi Ahh, good point.  I just updated the code as you described